### ISSUE TITLE
False positive of extension_access_modifier on property with private(set)

### ISSUE THREAD

**Issue by CraigSiemens** [realm/SwiftLint#5623]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When a `public` extension has a single property with `private(set)`, the property receives a warning that it should use an extension access modifier.

This isn't possible since private(set) cannot be moved from the property to the extension.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Linting Swift files in current working directory
Linting 'test.swift' (1/1)
test.swift:3:5: warning: Extension Access Modifier Violation: Prefer to use extension access modifiers (extension_access_modifier)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.55.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? homebrew
* Paste your configuration file: n/a

```yml
opt_in_rules:
  - extension_access_modifier
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? 15.4
* Do you have a sample that shows the issue? Yes

```swift
public class Foo {}
public extension Foo {
    private(set) var value: Int {
        get { 1 }
        set {}
    }
}
```

#### Single line to run it

```shell
echo "class Foo {}\npublic extension Foo {\n    private(set) var value {\n        get { 0 }\n        set { print(newValue) }\n    }\n}" | swiftlint lint --no-cache --use-stdin --enable-all-rules  
```
